liquibase-standard/src/main/java/liquibase/snapshot/jvm/UniqueConstraintSnapshotGenerator.java

```diff
@@ -166,148 +164,125 @@ protected List<CachedRow> listConstraints(Table table, DatabaseSnapshot snapshot
-                            + "and const.constraint_name='" + database.correctObjectName(name, UniqueConstraint.class) + "'";
+                    sql.append(String.format("and const.table_name='%s' ", database.correctObjectName(example.getRelation().getName(), Table.class)))
+                            .append(String.format("and const.constraint_name='%s' ", database.correctObjectName(name, UniqueConstraint.class)));
                 }
-                sql += "order by ordinal_position";
-                stmt = new RawSqlStatement(sql);
+                sql.append("order by ordinal_position");
+                rawSql = sql.toString();
             } else if (database instanceof PostgresDatabase) {
                 List<String> conditions = new ArrayList<>();
-                String sql = "select const.CONSTRAINT_NAME, COLUMN_NAME, const.constraint_schema as CONSTRAINT_CONTAINER "
-                        + "from " + database.getSystemSchema() + ".table_constraints const "
-                        + "join " + database.getSystemSchema() + ".key_column_usage col "
-                        + "on const.constraint_schema=col.constraint_schema "
-                        + "and const.table_name=col.table_name "
-                        + "and const.constraint_name=col.constraint_name ";
+                StringBuilder sql = new StringBuilder("select const.CONSTRAINT_NAME, COLUMN_NAME, const.constraint_schema as CONSTRAINT_CONTAINER ")
+                        .append(String.format("from %s.table_constraints const ", database.getSystemSchema()))
+                        .append(String.format("join %s.key_column_usage col ", database.getSystemSchema()))
+                        .append("on const.constraint_schema=col.constraint_schema ")
+                        .append("and const.table_name=col.table_name ")
+                        .append("and const.constraint_name=col.constraint_name ");
                 if (schema.getCatalogName() != null) {
-                    conditions.add("const.constraint_catalog='" + database.correctObjectName(schema.getCatalogName(), Catalog.class) + "'");
+                    conditions.add("const.constraint_catalog=?");
                 }
                 if (database instanceof CockroachDatabase) {
                     conditions.add("(select count(*) from (select indexdef from pg_indexes where schemaname='\" + database.correctObjectName(schema.getSchema().getName(), Schema.class) + \"' AND indexname='\" + database.correctObjectName(name, UniqueConstraint.class) + \"' AND (position('DESC,' in indexdef) > 0 OR position('DESC)' in indexdef) > 0))) = 0");
                     conditions.add("const.constraint_name != 'primary'");
                 }
                 if (schema.getSchema().getName() != null) {
-                    conditions.add("const.constraint_schema='" + database.correctObjectName(schema.getSchema().getName(), Schema.class) + "'");
+                    conditions.add("const.constraint_schema=?");
```

Comment:
Did those parameters were moved to line 211? As they are added conditionally should not they be defined around here in those if's?